src/modules/ce/components/AssignContactsForm.tsx

```diff
@@ -0,0 +1,86 @@
+import { Stack } from '@mui/material';
+import { reduce } from 'lodash-es';
+import React, { useCallback, useMemo, useState } from 'react';
+import AssignContactFormItem from './AssignContactFormItem';
+import LoadingButton from '@/components/elements/LoadingButton';
+import {
+  CeReferralFieldsFragment,
+  PickListOption,
+  useAssignParticipantsMutation,
+} from '@/types/gqlTypes';
+
+interface Props {
+  referral: CeReferralFieldsFragment;
+  projectId: string;
+  onClose?: VoidFunction;
+}
+const AssignContactsForm: React.FC<Props> = ({
+  referral,
+  projectId,
+  onClose,
+}: Props) => {
+  const initialValues = useMemo(() => {
+    return reduce(
+      referral.swimlanes,
+      (acc, swimlane) => {
+        acc[swimlane.id] = swimlane.participants.map((user) => {
+          return {
+            code: user.id,
+            label: user.name,
+          };
+        });
+        return acc;
+      },
+      {} as Record<string, PickListOption[]>
+    );
+  }, [referral.swimlanes]);
+
+  // `values` maps swimlaneId to PickListOption[]
+  // where PickListOption looks like: { code: <user ID>, label: <user name> }
+  const [values, setValues] =
+    useState<Record<string, PickListOption[]>>(initialValues);
+
+  const handleChangeValue = useCallback(
+    (swimlaneId: string, users: PickListOption[]) => {
+      setValues((prevValues: Record<string, PickListOption[]>) => ({
+        ...prevValues,
+        [swimlaneId]: users,
+      }));
+    },
+    [setValues]
+  );
+
+  const [handleSubmit, { loading, error }] = useAssignParticipantsMutation({
+    variables: {
+      referralId: referral.id,
+      // On submit, map the `values` object into the shape the mutation is expecting: [{ userId, swimlaneId }]
+      participants: Object.entries(values).flatMap(([swimlaneId, userIds]) =>
+        userIds.map(({ code }) => ({ userId: code, swimlaneId }))
+      ),
+    },
+    onCompleted: () => onClose?.(),
+  });
+
+  if (error) throw error;
+
+  return (
+    <Stack gap={2} mt={2}>
+      {referral.swimlanes?.map((swimlane) => (
+        <AssignContactFormItem
```

Comment:
I'm able to fully clear a swimlane. It seems like when I do that, and then a future step with no assignees should become available, it still remains locked  (the backend returns an 
initialized unavailable step).

Are you able to repro that? Maybe it has nothing to do with empty swimlanes, and is some other issue with how my workflow is set up. I'm using the "Enrollment Creator" template from the starter pack.

Repro steps:
- assign contants
- clear contacts
- perform move-in date step
- next step doesn't become available

Reply:
Nevermind, I couldn't reproduce it myself. It was an old referral I had sitting around. 
And I remember we have https://github.com/open-path/Green-River/issues/7615